src/compute/arithmetics/timestamp.rs

```diff
@@ -0,0 +1,403 @@
+/// Adds a duration to a timestamp array. The timeunit enum is used to scale
+/// correctly both arrays; adding seconds with seconds, or milliseconds with
+/// milliseconds.
+pub fn add_duration(
+    timestamp: &PrimitiveArray<i64>,
+    duration: &PrimitiveArray<i64>,
+) -> Result<PrimitiveArray<i64>> {
+    // Matching on both data types from both arrays.
+    // The timestamp and duration have a Timeunit enum in its data type.
+    // This enum is used to describe the addition of the duration.
+    match (timestamp.data_type(), duration.data_type()) {
+        (DataType::Timestamp(timeunit_a, _), DataType::Duration(timeunit_b)) => {
+            // Closure for the binary operation. The closure contains the scale
+            // required to add a duration to the timestamp array.
+            let scale = timeunit_scale(timeunit_a, timeunit_b);
+            let op = move |a, b| a + (b as f64 * scale) as i64;
+
+            binary(timestamp, duration, timestamp.data_type().clone(), op)
+        }
+        _ => Err(ArrowError::InvalidArgumentError(
+            "Incorrect data type for the arguments".to_string(),
+        )),
+    }
+}
+
+/// Subtract a duration to a timestamp array. The timeunit enum is used to scale
+/// correctly both arrays; subtracting seconds with seconds, or milliseconds with
+/// milliseconds.
+pub fn subtract_duration(
+    timestamp: &PrimitiveArray<i64>,
+    duration: &PrimitiveArray<i64>,
+) -> Result<PrimitiveArray<i64>> {
+    // Matching on both data types from both arrays.
+    // The timestamp and duration have a Timeunit enum in its data type.
+    // This enum is used to describe the addition of the duration.
+    match (timestamp.data_type(), duration.data_type()) {
+        (DataType::Timestamp(timeunit_a, _), DataType::Duration(timeunit_b)) => {
+            // Closure for the binary operation. The closure contains the scale
+            // required to add a duration to the timestamp array.
+            let scale = timeunit_scale(timeunit_a, timeunit_b);
+            let op = move |a, b| a - (b as f64 * scale) as i64;
+
+            binary(timestamp, duration, timestamp.data_type().clone(), op)
+        }
+        _ => Err(ArrowError::InvalidArgumentError(
+            "Incorrect data type for the arguments".to_string(),
+        )),
+    }
+}
+
+/// Calculates the difference between two timestamps returning an array of type
+/// Duration. The timeunit enum is used to scale correctly both arrays;
+/// subtracting seconds with seconds, or milliseconds with milliseconds.
+pub fn timestamp_diff(
+    lhs: &PrimitiveArray<i64>,
+    rhs: &PrimitiveArray<i64>,
+) -> Result<PrimitiveArray<i64>> {
+    // Matching on both data types from both arrays.
+    // Both timestamps have a Timeunit enum in its data type.
+    // This enum is used to adjust the scale between the timestamps.
+    match (lhs.data_type(), rhs.data_type()) {
+        (DataType::Timestamp(timeunit_a, _), DataType::Timestamp(timeunit_b, _)) => {
```

Comment:
I am not sure, I think that this is only valid for naive (i.e. `None`) timezones. For time-aware, I think that we need to take into account the summer times and all that complexity here.

Reply:
mmm... that's right. I think it will need to use crono to consider the timezones. To make this difference time-aware we would have to create a larger PR. 

I created this function thinking it may be useful to have it, but if you think we should wait until we have better management of timezones, I can remove it.

Reply:
(I like it, I would just use `None` there for now, and handle the timezone problems in another PR)

Reply:
Cool. I left only the case when both timezones are `None` and that is tested as well. This could be the template function to work with more complex timezones.